pt/controller_1/controller-1.py

```diff
@@ -108,200 +192,124 @@ def initialise():
+
+def readings(moisture_0, moisture_1, moisture_2, sht45, light_sensor):
+    print(f"Sample at: {datetime.now().isoformat()}")
+    print_sht45_measurements(measurements = sht45.measurements)
+    point = create_sht45_point_measurement(measurements = sht45.measurements)
     
     store_influx.write(record=point)
 
-    print("AS7341 light sensor: 415nm wavelength (Violet)  %s" % light_sensor.channel_415nm)
-    print("AS7341 light sensor: 445nm wavelength (Indigo) %s" % light_sensor.channel_445nm)
-    print("AS7341 light sensor: 480nm wavelength (Blue)   %s" % light_sensor.channel_480nm)
-    print("AS7341 light sensor: 515nm wavelength (Cyan)   %s" % light_sensor.channel_515nm)
-    print("AS7341 light sensor: 555nm wavelength (Green)   %s" % light_sensor.channel_555nm)
-    print("AS7341 light sensor: 590nm wavelength (Yellow)  %s" % light_sensor.channel_590nm)
-    print("AS7341 light sensor: 630nm wavelength (Orange)  %s" % light_sensor.channel_630nm)
-    print("AS7341 light sensor: 680nm wavelength (Red)     %s" % light_sensor.channel_680nm)
-    point = (
-        Point("as7341")
-        .field("violet", light_sensor.channel_415nm)
-        .field("indigo", light_sensor.channel_445nm)
-        .field("blue", light_sensor.channel_480nm)
-        .field("cyan", light_sensor.channel_515nm)
-        .field("green", light_sensor.channel_555nm)
-        .field("yellow", light_sensor.channel_590nm)
-        .field("orange", light_sensor.channel_630nm)
-        .field("red", light_sensor.channel_680nm)
-    )
+    print_as7341_measurements(light_sensor = light_sensor)
+    point = create_as7341_point_measurement(light_sensor = light_sensor)
     
     store_influx.write(record=point)
-    print("light sensor read")
 
     moisture_reading = moisture_0.moisture_read()
     temperature_reading = moisture_0.get_temp()
-    print(f"Soil sensor-0 --> Moisture: {moisture_reading:0.1f}, Temperature: {temperature_reading:0.1f} C")
-    point = (
-        Point("soil-sensor-0")
-        .field("moisture", moisture_reading)
-        .field("temperature", temperature_reading)
-    )
+    point = create_soil_sensor_point_measurement(moisture_reading = moisture_reading, temperature_reading = temperature_reading, sensor_id = 0)
+    print_soil_sensor_measurements(moisture_reading = moisture_reading, temperature_reading = temperature_reading, sensor_id = 0)
     
     store_influx.write(record=point)
     
     moisture_reading = moisture_1.moisture_read()
     temperature_reading = moisture_1.get_temp()
-    print(f"Soil sensor-1 --> Moisture: {moisture_reading:0.1f}, Temperature: {temperature_reading:0.1f} C")
-    point = (
-        Point("soil-sensor-1")
-        .field("moisture", moisture_reading)
-        .field("temperature", temperature_reading)
-    )
+    point = create_soil_sensor_point_measurement(moisture_reading = moisture_reading, temperature_reading = temperature_reading, sensor_id = 1)
```

Comment:
the create and print functions are calling the sensor. Thus the measurement being stored in influx is not the same one being printed. It is better to capture the measurement once and pass it to both create and print functions. Please make the necessary changes to similar calls in the later parts of the code.